src/sparse/vec.rs

```diff
@@ -486,27 +486,150 @@ where
+        indices: IStorage,
+        data: DStorage,
+    ) -> Result<Self, (IStorage, DStorage, StructureError)> {
+        if I::from(n).is_none() {
+            return Err((
+                indices,
+                data,
+                StructureError::OutOfRange("Index size is too small"),
+            ));
+        }
+        if indices.len() != data.len() {
+            return Err((
+                indices,
+                data,
+                StructureError::SizeMismatch(
+                    "indices and data do not have compatible lengths",
+                ),
+            ));
+        }
+        for i in indices.iter() {
+            if i.to_usize().is_none() {
+                return Err((
+                    indices,
+                    data,
+                    StructureError::OutOfRange(
+                        "index can not be converted to usize",
+                    ),
+                ));
+            }
+        }
+        if !utils::sorted_indices(indices.as_ref()) {
+            return Err((
+                indices,
+                data,
+                StructureError::Unsorted("Unsorted indices"),
+            ));
+        }
+        if let Some(i) = indices.last() {
+            if i.to_usize().unwrap() >= n {
+                return Err((
+                    indices,
+                    data,
+                    StructureError::SizeMismatch(
+                        "indices larger than vector size",
+                    ),
+                ));
+            }
+        }
+        Ok(Self::new_trusted(n, indices, data))
+    }
+
+    /// Create a sparse vector
     ///
     /// # Panics
     ///
     /// - if `indices` and `data` lengths differ
     /// - if the vector contains out of bounds indices
-    pub fn new(n: usize, indices: Vec<I>, data: Vec<N>) -> Self
-    where
-        N: Copy,
-    {
-        Self::try_new(n, indices, data).unwrap()
+    /// - if indices are out of order
+    ///
+    /// # Examples
+    /// ```rust
+    /// # use sprs::*;
+    /// // Creating a sparse owned vector
+    /// let owned = CsVec::new(10, vec![0, 4], vec![-4, 2]);
+    /// // Creating a sparse borrowing vector with `I = u16`
+    /// let borrow = CsVecViewI::new(10, &[0_u16, 4], &[-4, 2]);
+    /// // Creating a general sparse vector with different storage types
+    /// let mixed = CsVecBase::new(10, &[0_u64, 4] as &[_], vec![-4, 2]);
+    /// ```
+    pub fn new(n: usize, indices: IStorage, data: DStorage) -> Self {
+        Self::try_new(n, indices, data)
+            .map_err(|(_, _, e)| e)
+            .unwrap()
     }
 
-    /// Try create an owning `CsVec` from vector data.
+    /// Try create a sparse vector
     pub fn try_new(
         n: usize,
-        indices: Vec<I>,
-        data: Vec<N>,
-    ) -> Result<Self, StructureError>
+        indices: IStorage,
+        data: DStorage,
+    ) -> Result<Self, (IStorage, DStorage, StructureError)> {
+        Self::new_(n, indices, data)
+    }
```

Comment:
Looks like `try_new` is actually exactly `new_`, so maybe we can remove `new_` now.

Reply:
Good idea! Done